src/ressources/customers/CustomerEdit.tsx

```diff
@@ -0,0 +1,148 @@
+import { Box, Card, Grid, Typography } from "@mui/material";
+import {
+  DateInput,
+  Edit,
+  EditProps,
+  NullableBooleanInput,
+  PasswordInput,
+  SelectArrayInput,
+  SimpleForm,
+  TextInput,
+  useRecordContext,
+  UseRecordContextParams,
+} from "react-admin";
+import { FieldErrors, FieldValues } from "react-hook-form";
+import { Aside } from "./Aside";
+import { Customer } from "./Customers";
+
+const CustomerTitle = (props: UseRecordContextParams<Customer>) => {
+  const record = useRecordContext<Customer>(props);
+  return (
+    <Typography>
+      {record.first_name} {record.last_name}
+    </Typography>
+  );
+};
+
+const validateCustomerEdit = (values: FieldValues) => {
+  const errors: FieldErrors = {};
+  if (!values.first_name) {
+    errors.first_name = "The firstName is required";
+  }
+  if (!values.last_name) {
+    errors.last_name = "The lastName is required";
+  }
+  if (!values.email) {
+    errors.email = "The email is required";
+  }
```

Comment:
Why do you need this code? Is it a test you made to see if you could override the default behaviour?